src/adapter/philips_hue/mod.rs

```diff
@@ -0,0 +1,273 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate serde_json;
+
+mod http;
+mod nupnp;
+mod api;
+
+use adapter::philips_hue::api::structs::*;
+use adapter::philips_hue::api::HueLight;
+use controller::Controller;
+use events::*;
+use iron::{ Request, Response, IronResult };
+use iron::headers::ContentType;
+use iron::status::Status;
+use iron::method::Method;
+use router::Router;
+use service::{ Service, ServiceAdapter, ServiceProperties };
+use std::time::Duration;
+use std::thread;
+use uuid::Uuid;
+use std::io::Read;
+
+
+pub struct PhilipsHueAdapter<T> {
+    name: String,
+    controller: T,
+    hubs: Vec<api::HueApi>,
+}
+
+impl<T: Controller> PhilipsHueAdapter<T> {
+    pub fn new(controller: T) -> Self {
+        debug!("Creating Philips Hue adapter");
+        PhilipsHueAdapter { name: "PhilipsHueAdapter".to_owned(),
+                       controller: controller,
+                       hubs: nupnp::query(),
```

Comment:
Originally the query was inside the thread, but that resulted in and Arc weirdness and lifetime carnage in other structs used inside the hub loop. The fix was to make the ServiceAdapter static, resulting in this block. Luckily I don't need the hubs to be part of the ServiceAdapter's state any longer due to design changes towards static Hue objects in general. Trivial fix. Done. For now...

At some point though the ServiceAdapter and many other structs can't be static any longer. I have been down that road and there were lifetime roadblocks every meter. Can you suggest someone to help out with that?